### ISSUE TITLE
issues with multer

### ISSUE THREAD

**Issue by tsm91** [expressjs/multer#207]
Hello, I am using Multer with GraphicsMagick and ran into a few issues.

1, I am experiencing that whenever an error occurs and i redirect the user, the browser will hang and never redirect.

2, for some reason fileSize limit doesn't work, on big images. I tested this by uploading a 14mb JPG image, I didnt get back a `LIMIT_FILE_SIZE` error object, but `req.file` was undefined.

3, how can i remove the temp files from `dest` path? They don't seem to be removed automatically after upload finishes.

The code:

image-uploader.js (multer related part)
```js
    upload: multer({
      dest: path.join('.tmp', 'uploads'),
      limits: {
        files: 1,
        fileSize: 5 * 1024 * 1024
      },
      fileFilter: function(req, file, cb) {
        var type = file.mimetype;
        if (type !== 'image/png' && type !== 'image/jpg' && type !== 'image/jpeg') {
          cb(new Error(lang.error.invalidImgType));
        } else {
          cb(null, true);
        }
      }
    }),
```

router.js
```js
function uploadImage(req, res, next) {
    var upload = IMU.upload.single('avatar');

    upload(req, res, function(err) {
      if (err) {
        if (err.code === 'LIMIT_FILE_SIZE') err.message = lang.error.minImgSize;

        req.flash('error', err.message);
        return res.redirect('/en');
      }

      if (_.isUndefined(req.file)) {
        req.flash('error', lang.error.noImageUploaded);
        return res.redirect('/en');
      }

      IMU.handleImage(req.file, req.user.name, function(err, image) {
        if (err) {
          req.flash('error', err.message);
          return res.redirect('/en');
        }

        res.render('my-account', {
          status: 'Finished uploading',
          newImage: image
        });
      });
    })
  }
```

**Comment by LinusU** [comment]
@tsm91 This is very possibly related to a stream bug in Node.js itself (nodejs/node#2323)

Could you please test the code in #205 and see if that solves the problems for you?

**Comment by tsm91** [comment]
@LinusU I tested it, i couldn't reproduce the browser hang. Tested on `v0.12.7`

In my case this fixes the first (and most serious) problem.

Any idea for the 3rd one? I saw there is a bug for my 2nd issue too

**Comment by kadishmal** [comment]
Today I upgraded `multer` from 0.x to 1.0.3 and encountered the same error. The test passes in Node 0.10.38 but fails in 0.10.30.

What I have noticed is that `busboy` successfully detects that the limit has been hit, and successfully emits the `limit` event. `multer` successfully receives this event and is about to call the callback, however, `bosboy` seems to fire the `finish` event before `multer` is able to call the `abortWithError` function.
